src/rust-crypto/RoomEncryptor.ts

```diff
@@ -103,7 +110,9 @@ export class RoomEncryptor {
         this.prefixedLogger.debug("Sessions for users are ready; now sharing room key");
 
         const rustEncryptionSettings = new EncryptionSettings();
-        /* FIXME historyVisibility, rotation, etc */
+        /* FIXME rotation, etc */
+
+        rustEncryptionSettings.historyVisibility = toRustHistoryVisibility(this.room.getHistoryVisibility());
```

Comment:
we should either make this use `this.encryptionSettings`, or get rid of `this.encryptionSettings`.

I can't really remember why `this.encryptionSettings` exists, but I think it might be useful when we come to implement https://github.com/vector-im/element-web/issues/26108.

Reply:
According to [this comment](https://github.com/matrix-org/matrix-js-sdk/blob/florianduros/elementr/wire-up-history-visibility/src/rust-crypto/RoomEncryptor.ts#L46), the `encryptionSettings` is the content of the [`m.room.encryption`](https://spec.matrix.org/latest/client-server-api/#mroomencryption) event. I'm wiring up the `encryptionSettings` in this PR https://github.com/matrix-org/matrix-js-sdk/pull/3807.

Since the history visibility is not a part of the `m.room.encryption` event, it seems to me a bit odd to put it in it.

Reply:
oh right. Yes, sorry, ignore me.